src/MultiRegion/cubed_sphere_boundary_conditions.jl

```diff
@@ -0,0 +1,376 @@
+function fill_cubed_sphere_halo_regions!(field::CubedSphereField{<:Center, <:Center})
+    grid = field.grid
+
+    if !(grid isa ConformalCubedSphereGrid)
+        return
+    end
```

Comment:
what is this doing?
is there a chance `field.grid` is not a ConformalCubedSphereGrid when `field::CubedSphereField`?